harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -762,47 +835,72 @@ def create_lr_scheduler(
-        device (slot) since the start of training.
-        """
+        Train on one batch.
+
+        Users should implement this function by doing the following things:
+        1. Run forward passes on the models.
+
+        2. Calculate the gradients with the losses with ``backward``.
+
+        3. Call an optimization step for the optimizers with ``step_optimizer``. You can clip
+           gradients by specifying the argument ``clip_grads``.
+
+        4. Step LR schedulers if using manual step mode.
+
+        5. Return training metrics in a dictionary.
+
+        Here is a code example.
 
-        # TODO(DET-3204): Add the following comments to the docstring.
-        # If the models, optimizers, and LR schedulers are initialized in :meth:`__init__`
-        # by users, users should calculate the gradients with the losses with :meth:`backward`
-        # and call an optimization step for the optimizers with :meth:`step_optimizer`
-        # on your own. Here is a code example.
-        #
-        # .. code-block:: python
-        #
-        #     # Assume having initialized models, optimizers, and LR schedulers (when
-        #     # using manual mode): self.model1, self.model2, self.opt1, self.opt2, and self.lrs1.
-        #
-        #     # Calculate the losses using the models.
-        #     loss1 = self.model1(batch)
-        #     loss2 = self.model2(batch)
-        #
-        #     # Run backward passes on losses and step optimizers. These can happen
-        #     # in arbitrary orders.
-        #     self.context.backward(loss1)
-        #     self.context.backward(loss2)
-        #     self.context.step_optimizer(self.opt1)
-        #     self.context.step_optimizer(self.opt2)
-        #
-        #     # Step the learning rate.
-        #     self.lrs1.step()
-        #
-        #     return {"loss1": loss1, "loss2": loss2}
-
-        # TODO(DET-3267): deprecate the model argument when releasing pytorch flexible primitives.
+        .. code-block:: python
+
+            # Assume having initialized models, optimizers, and LR schedulers (when
+            # using manual mode): self.model1, self.model2, self.opt1, self.opt2, and self.lrs1.
+
+            # Calculate the losses using the models.
+            loss1 = self.model1(batch)
+            loss2 = self.model2(batch)
+
+            # Run backward passes on losses and step optimizers. These can happen
+            # in arbitrary orders.
+            self.context.backward(loss1)
+            self.context.backward(loss2)
+            self.context.step_optimizer(
+                self.opt1,
+                clip_grads=lambda params: torch.nn.utils.clip_grad_norm_(params, 0.0001),
+            )
+            self.context.step_optimizer(self.opt2)
+
+            # Step the learning rate.
+            self.lrs1.step()
+            self.lrs2.step()
+
+            return {"loss1": loss1, "loss2": loss2}
```

Comment:
Question: Do you know what do we render in the UI if we don't return a key named "loss"?  Do we plot all the training metrics on one graph?